### ISSUE TITLE
Select serie from horizontal bottom legend, select other serie.

### ISSUE THREAD

**Issue by fertegli** [highcharts/highcharts#21994]
#### Expected behaviour
When clicking on a series in the chart legend, the corresponding series should be selected or deselected accurately, and the chart should update accordingly. This behavior should work consistently across different web browsers, even when multiple series (e.g., 10 or more) are plotted.

#### Actual behaviour
In Microsoft Edge, when clicking on a series in the chart legend, the wrong series is selected or deselected. The selected series is often misaligned, and the click affects a series two lines below the one that was clicked. This issue occurs when multiple series (e.g., 10 or more) are plotted. In other browsers like Google Chrome, the selection works as expected.

Steps to reproduce:
1. Click on Series 1 in the chart legend. Remember work in Microsoft EDGE.
2. Move the cursor close to the "-" symbol (the part of the legend that represents the series line).
3. Subtly shift the cursor slightly upward without moving too far from the legend item.
4. Click, and notice that Series 7 gets selected or deselected instead of Series 1.

#### Live demo with steps to reproduce
https://jsfiddle.net/kwb6mxda/

#### Product version
Highcharts Core v11.4.8 (but I tested on v11.1.0 too).

#### Affected browser(s)
Microsoft EDGE

**Comment by karolkolodziej** [comment]
Hi @fertegli!

I've tested it with `Microsoft Edge Version 129.0.2792.65 (Official build) (64-bit)` and I don't see the issue.
Please check the recording. 

https://github.com/user-attachments/assets/cdb68e64-3504-4093-90ad-fa819e119208

What am I missing here?

**Comment by fertegli** [comment]
> Hi @fertegli!
> 
> I've tested it with `Microsoft Edge Version 129.0.2792.65 (Official build) (64-bit)` and I don't see the issue. Please check the recording.
> 
>  screenRecording-14-10-2024-10-5.mp4 
> What am I missing here?

Hi Karol,

I forgot to mention that for the issue to occur, there needs to be several legends stacked vertically. The misalignment happens when there are enough series for the legends to form multiple rows.

Additionally, I have recorded a video demonstrating the issue, which I’m attaching as evidence.

I used Microsoft Edge versión 129.0.2792.89 (Compilación oficial) (64 bits)

Let me know if you need any further details.

Best regards,
FerTegli

https://github.com/user-attachments/assets/050dd715-9de7-4680-af89-bc64a6957acb

**Comment by fertegli** [comment]
Hi Karol. 
Thanks a lot for your interest in this bug.
This was reported by users, working on Win 11 PC. I can recreate that in my environment. I don't use any plugins. 
I made another video, debuging to show how EDGE create controls in wrong place, but only with  more than 5 legend in the same column (Chrome don't do it in the same way, but in this customer, the don't have Chrome in standard workstations). In the video, you could see how controls are misplaced in legend. Text is in spanish, but you could see how some controls are placed to line up in legend.

https://github.com/user-attachments/assets/5f96fca3-e8c2-4ecb-a16f-a95fb4ee6e8d

Thanks in advance for you help.

**Comment by KacperMadej** [comment]
Thank you for all the material that helped with recreating the problem.
Looks like a bug in a11y.

This can be recreate in Chrome with Highcharts Core and a11y module only.
It is easier to recreate when the chart position require some vertical scrolling.
Demo: https://jsfiddle.net/BlackLabel/bc8vajsx/

Steps:
1. Scroll down to the chart - if you don't need to scroll down, adjust height of browser or demo preview window.
2. Hide some series.
3. Check the layout of the container with `highcharts-a11y-proxy-group highcharts-a11y-proxy-group-legend` classes.

![image](https://github.com/user-attachments/assets/2cad44d9-dde1-4bfa-a11d-00d8aac0260d)

_Internal note:_
This is a visible problem when the offset from the correct position makes the proxy element land on top of something else but sometimes the offset can be big enough to push the elements outside of the legend or even the chart.
